src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -192,26 +223,43 @@ public FormValidation doCheckReturnStatus(@QueryParameter boolean returnStdout,
                     LOGGER.log(Level.FINE, "Jenkins is not running, no such node {0}, or it is offline", node);
                     return null;
                 }
+                if (watching) {
+                    try {
+                        controller.watch(ws, new HandlerImpl(this, ws, listener()), listener());
+                        recurrencePeriod = WATCHING_RECURRENCE_PERIOD;
+                    } catch (UnsupportedOperationException x) {
+                        getContext().onFailure(x);
```

Comment:
Would it be useful to set `watching = false` and fall back to polling in this case? Is there even any way to get here without modifying the serialized state? I guess if the controller implementation supported `watch`, and then we changed the implementation and removed the override, then any pipelines that resume after upgrading this plugin would break, but that feels like a pretty contrived case. Maybe worth a comment that this should never happen IIUC.

Reply:
Yes it should never happen; can make that clearer in a comment.